### DISCUSSION TITLE
to_networkx() arguments require iterable of str

### DISCUSSION THREAD

**Discussion by jackychen08** [pyg-team/pytorch_geometric#7987] [Q&A]
Currently, to_networkx() arguments such as node_attrs and edge_attrs require " iterable of str". This seems unintuitive as lots of features are stored as floats or ints in pytorch tensors. Would it make more sense to update this function to be able to accept tensors of floats/ints ?

**Comment by rusty1s** [comment]
This just refers to the names of attributes stored as part of `Data`, so I am not sure how one could support floats or ints here. Can you clarify?

**Reply by jackychen08** [reply]
From reading the documentation [here ](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.to_networkx), it looks like node_attrs, edge_attrs, etc should be passed in separately. Ie, the function call would look like this:
`to_networkx(data, node_attrs = data.node_attrs, edge_attrs = data.edge_attrs)`. Do I not need to pass these attributes explicitly?

**Reply by rusty1s** [reply]
No, this looks fine. `data.node_attrs` and `data.edge_attrs` are of type `List[str]`, so this matches with the function signature of `to_networkx`. Can you clarify your issue for me?
